cmd/sovereignnode/chainSimulator/tests/bridge/bridge_test.go

```diff
@@ -0,0 +1,221 @@
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, esdtSafeAddress, big.NewInt(0), setSovereignBridgeAddressData, uint64(10000000))
+
+	supply, _ := big.NewInt(0).SetString("123000000000000000000", 10)
+	issueArgs := "issue" +
+		"@" + hex.EncodeToString([]byte("MainToken")) +
+		"@" + hex.EncodeToString([]byte("TKN")) +
+		"@" + hex.EncodeToString(supply.Bytes()) +
+		"@" + fmt.Sprintf("%X", 18) // num decimals
+	issueTx := chainSimulatorIntegrationTests.SendTransactionAndWaitForMaxNrOfBlocks(t, cs, wallet.Bytes, &nonce, systemEsdtScAddress, issueCost, issueArgs, uint64(60000000), 3) // 3 blocks to execute on destination shard (meta)
+	tokenIdentifier := issueTx.Logs.Events[0].Topics[0]
+	require.True(t, len(tokenIdentifier) > 7)
+
+	err = cs.GenerateBlocks(1) // one more block to get the tokens in source shard
+	require.Nil(t, err)
+
+	tokens, _, err := cs.GetNodeHandler(0).GetFacadeHandler().GetAllESDTTokens(wallet.Bech32, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.NotNil(t, tokens)
+	require.True(t, len(tokens) == 1)
+	require.Equal(t, supply.String(), tokens[string(tokenIdentifier)].GetValue().String())
+
+	amountToDeposit, _ := big.NewInt(0).SetString("2000000000000000000", 10)
+	depositArgs := "MultiESDTNFTTransfer" +
+		"@" + hex.EncodeToString(esdtSafeAddress) +
+		"@01" +
+		"@" + hex.EncodeToString(tokenIdentifier) +
+		"@" + // nonce
+		"@" + hex.EncodeToString(amountToDeposit.Bytes()) +
+		"@" + hex.EncodeToString([]byte("deposit")) + // deposit func
+		"@" + hex.EncodeToString(wallet.Bytes) //receiver from other side
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, wallet.Bytes, big.NewInt(0), depositArgs, uint64(20000000))
+
+	tokens, _, err = cs.GetNodeHandler(0).GetFacadeHandler().GetAllESDTTokens(wallet.Bech32, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.NotNil(t, tokens)
+	require.True(t, len(tokens) == 1)
+	require.Equal(t, big.NewInt(0).Sub(supply, amountToDeposit).String(), tokens[string(tokenIdentifier)].GetValue().String())
+}
+
+func TestBridge_DeployOnSovereignChain_IssueAndDeposit(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	scs, err := sovereignChainSimulator.NewSovereignChainSimulator(sovereignChainSimulator.ArgsSovereignChainSimulator{
+		SovereignConfigPath: sovereignConfigPath,
+		ChainSimulatorArgs: &chainSimulator.ArgsChainSimulator{
+			BypassTxSignatureCheck: false,
+			TempDir:                t.TempDir(),
+			PathToInitialConfig:    defaultPathToInitialConfig,
+			NumOfShards:            1,
+			GenesisTimestamp:       time.Now().Unix(),
+			RoundDurationInMillis:  uint64(6000),
+			RoundsPerEpoch:         core.OptionalUint64{},
+			ApiInterface:           api.NewNoApiInterface(),
+			MinNodesPerShard:       2,
+			ConsensusGroupSize:     2,
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, scs)
+
+	defer scs.Close()
+
+	err = scs.GenerateBlocks(200)
+	require.Nil(t, err)
```

Comment:
```suggestion
	err = scs.GenerateBlocks(200)
	require.Nil(t, err)
```
+ also same comments with functions and so forth.